docs/metadata.md

```diff
@@ -0,0 +1,264 @@
+
+```yaml
+# freshmint.yaml
+schema:
+  fields:
+    - name: city
+      type: string
+    - name: description
+      type: string
+    - name: image
+      type: ipfs-file
+    - name: serial
+      type: uint64
+  views:
+    - type: display
+      options:
+        name: name
+        description: description
+        # Link to the "image" field above
+        thumbnail: image
+        
+    - type: serial
+      options:
+        # Link to the "serial" field above
+        serialNumber: serial
+    
+    - type: external-url
+      options: "${collection.url}/nfts/${nft.owner}/${nft.id}"
+      
+    # These fields don't require any options
+    - type: nft
+    - type: nft-collection-display
+    - type: nft-collection-data
+```
+
+## Customize your metadata schema
+
+Every new Freshmint project uses a schema with three simple fields: `name`, `description` and `thumbnail`.
+
+However, you can add and remove fields by updating `freshmint.yaml`.
+
+For example, you may want to attach a video file to each NFT:
+
+```yaml
+# freshmint.yaml
+contract:
+  name: Foo
+  type: standard
+  schema:
+    fields:
+      - name: name
+        type: string
+      - name: description
+        type: string
+      - name: thumbnail
+        type: ipfs-file
+      # Add a video field to each NFT
+      - name: video
+        type: ipfs-file
+    views:
+      ...
+```
+
+After updating your schema, you'll need to update the columns in `nfts.csv` to match the new schema.
+
+Here we add the video column:
+
+```csv
+name,description,thumbnail,video
+London,"A watercolor scene from London, UK.",london.png,london.mp4
+...
+```
+
+Next, use the `fresh gen` command to regenerate your Cadence contract.
+
+```sh
+fresh gen cadence
+```
+
+Lastly, restart the dev server to redeploy your contract.
+
+```sh
+fresh dev
+```
+
+## Working with media assets
+
+Your metadata can include links to external assets like images and videos.
+
+### IPFS assets
+
+> We recommend storing assets in [IPFS](https://ipfs.tech/), a peer-to-peer storage network.
+
+Start by adding an `ipfs-file` field to your schema.
+
+The default schema already stores `thumbnail` in IPFS:
+
+```yaml
+# freshmint.yaml
+contract:
+  name: Foo
+  type: standard
+  schema:
+    fields:
+      - name: name
+        type: string
+      - name: description
+        type: string
+      - name: thumbnail
+        type: ipfs-file # Thumbnails are stored in IPFS
+    views:
+      ...
+```
+
+#### Pin local files
+
+The easiest way to pin an IPFS asset is by moving it to your project's `assets` directory.
+
+```
+/your-project
+  /assets
+    foo.jpeg
+```
+
+Then you can reference the asset in your `nfts.csv` file:
+
+```csv
+name,description,thumbnail
+Foo,This is the foo NFT,foo.jpeg
+```
+
+Now when you run `fresh mint`, Freshmint will automatically pin the asset to IPFS and attach its
+[CID](https://docs.ipfs.tech/concepts/content-addressing/#what-is-a-cid) to the minted NFT.
```

Comment:
Freshmint uses the IPFS pinning service endpoint (pre-populated with the [NFT.Storage](https://nft.storage) [API URL](https://api.nft.storage)) and key (must be manually entered) defined in the .env in the root directory after creating a project.